kong/runloop/plugins_iterator.lua

```diff
@@ -287,6 +287,9 @@ local function get_next(self)
       local cfg = load_configuration_through_combos(ctx, combos, plugin)
       if cfg then
         plugins[name] = cfg
+        if plugin.handler.response and plugin.handler.response ~= BasePlugin.response then
+          self.iterator.has_response = true
```

Comment:
I fear that this might leak to other requests. E.g.:
1. user goes to route r
2. route r has plugin p1 and plugin p2
3. p1 has priority of 10 and p2 has priority of 1
4. p1 does not implement response, but p2 does
5. in request r1 the both plugins run, which set `iterator.has_response = true`
6. in request r2 only p1 runs and it shortcircuits, what is the value of `iterator.has_response`?
7. also as iterator remains the same, would it be forever true, once it has been set to true?

OR did I understand it wrong?

Reply:
AFAICT, the flag is persistent but has limited scope, hopefully just the route, so a request would be buffered if any plugin _in the route_ implements `:response`, but not other routes with a different plugin set.

i didn't consider the case where an "early" plugin shortcircuits.  in effect, that would make it impossible to cache.

I guess the choice is to consider the "buffering" to be statically or dynamically determined.  which would be the least surprising / easier to document?